src/core/core/rpc_engine.cpp

```diff
@@ -850,37 +850,35 @@ namespace dsn {
                             {
                                 resolver->on_access_failure(req2->header->gpid.u.partition_index, err);
 
-                                /*
-                                // TODO(qinzuoyan): because the user will fetch reply states from this response_task 'call',
-                                // so it's useless to create another response_task, which will not be handled by user.
-
                                 // still got time, retry
                                 uint64_t nms = dsn_now_ms();
-                                if (nms + 1000 < timeout_ts_ms)
+                                if (nms + 10 < timeout_ts_ms)
```

Comment:
And another proposal, maybe we can record timeout_ts_ms in message_header::client (using the old padding field), then we can avoid calculating it repeatedly, which is also not precise.
```
struct message_header
{
   ....
   struct
   {
       uint64_t hash; // for both partition hash and thread hash for the exact location of this request
       int32_t  timeout_ms;
       int32_t  timeout_ts_ms; // the time point of timeout
   } client;
   ....
};
```